pkg/imageengine/buildah/manifest.go

```diff
@@ -15,29 +15,177 @@
+
+	names, err := util.ExpandNames([]string{name}, systemCxt, store)
 	if err != nil {
+		return fmt.Errorf("encountered while expanding image name %q: %w", name, err)
+	}
+
+	imageID, err := list.SaveToImage(store, "", names, manifest.DockerV2ListMediaType)
+	if err == nil {
+		fmt.Printf("%s\n", imageID)
+	}
+
+	return err
+}
+
+func (engine *Engine) DeleteManifests(names []string, opts *options.ManifestDeleteOpts) error {
+	runtime := engine.ImageRuntime()
+
+	rmiReports, rmiErrors := runtime.RemoveImages(context.Background(), names, &libimage.RemoveImagesOptions{
+		Filters:        []string{"readonly=false"},
+		LookupManifest: true,
+	})
+	for _, r := range rmiReports {
+		for _, u := range r.Untagged {
+			fmt.Printf("untagged: %s\n", u)
+		}
+	}
+	for _, r := range rmiReports {
+		if r.Removed {
+			fmt.Printf("%s\n", r.ID)
+		}
+	}
+
+	var multiE *multierror.Error
+	multiE = multierror.Append(multiE, rmiErrors...)
+	return multiE.ErrorOrNil()
+}
+
+func (engine *Engine) InspectManifest(name string, opts *options.ManifestInspectOpts) error {
+	printManifest := func(manifest []byte) error {
+		var b bytes.Buffer
+		err := json.Indent(&b, manifest, "", "    ")
+		if err != nil {
+			return fmt.Errorf("rendering manifest for display: %w", err)
+		}
+
+		fmt.Printf("%s\n", b.String())
+		return nil
+	}
+
+	runtime := engine.ImageRuntime()
+	store := engine.ImageStore()
+	systemCxt := engine.SystemContext()
+	ctx := getContext()
+
+	// Before doing a remote lookup, attempt to resolve the manifest list
+	// locally.
+	manifestList, err := runtime.LookupManifestList(name)
+	if err == nil {
+		schema2List, err := manifestList.Inspect()
+		if err != nil {
+			return err
+		}
+
+		rawSchema2List, err := json.Marshal(schema2List)
+		if err != nil {
+			return err
+		}
+
+		return printManifest(rawSchema2List)
+	}
+	if !errors.Is(err, storage.ErrImageUnknown) && !errors.Is(err, libimage.ErrNotAManifestList) {
 		return err
 	}
 
-	manifestList, err := runtime.LookupManifestList(listImageSpec)
+	// TODO: at some point `libimage` should support resolving manifests
+	// like that.  Similar to `libimage.Runtime.LookupImage` we could
+	// implement a `*.LookupImageIndex`.
+	refs, err := util.ResolveNameToReferences(store, systemCxt, name)
+	if err != nil {
+		logrus.Debugf("error parsing reference to image %q: %v", name, err)
+	}
+
+	if ref, _, err := util.FindImage(store, "", systemCxt, name); err == nil {
+		refs = append(refs, ref)
+	} else if ref, err := alltransports.ParseImageName(name); err == nil {
+		refs = append(refs, ref)
+	}
+	if len(refs) == 0 {
+		return fmt.Errorf("locating images with names %v", name)
+	}
+
+	var (
+		latestErr error
+		result    []byte
+	)
+
+	appendErr := func(e error) {
+		if latestErr == nil {
+			latestErr = e
+		} else {
+			latestErr = fmt.Errorf("tried %v: %w", e, latestErr)
+		}
+	}
+
+	for _, ref := range refs {
+		logrus.Debugf("Testing reference %q for possible manifest", transports.ImageName(ref))
```

Comment:
I don't know what our principles for using `logrus` and `fmt` are?

Could you please make a simple explanation?